### ISSUE TITLE
Does it make sense to set benchmark=risk_free_rate in semicovariance?

### ISSUE THREAD

**Issue by schneiderfelipe** [robertmartin8/PyPortfolioOpt#36]
It seems that semicovariance, as implemented in pyportfolioopt, sets a penalty for assets that go below a certain threshold. Even though it's natural to demand this threshold to be non-negative, wouldn't it be more reasonable to give it a more meaningful value as default, e.g., the risk free rate, instead of zero?

Sorry if I'm missing something here, is there a reason for this being zero by default?

**Comment by robertmartin8** [comment]
Hi Felipe,

Great to hear from you again! You are entirely correct – the risk-free rate is a much more natural benchmark. Indeed, that's how the downside deviation (used in the Sortino ratio) is calculated, and really the semicovariance matrix is meant to generalise downside deviation in the same way that covariance generalises variance.

I was going to justify the default benchmark of zero in terms of API design, arguing that it would require me to hard code a risk-free rate. But I remembered that I actually have done that in the `EfficientFrontier` class, so it's actually inconsistent for me to set it at zero. 

Thanks for bringing this to my attention, I'll fix it now.

**Comment by schneiderfelipe** [comment]
Great! Thanks for the help! Just two more things.

*One*: since the benchmark is subtracted from (daily?) returns in some cases,

```python
drops = np.fmin(daily_returns - benchmark, 0)
```

but it's subtracted from (annual?) returns in other cases,

```python
(mu - risk_free_rate) / sigma
```

isn't there a discrepancy?

*Two*: I managed to use a market benchmark with `semicovariance`, I'm just not sure if my approach is sound enough:

```python
# daily_benchmark is a series with daily *returns* from a benchmark e.g. S&P500
# prices is a dataframe with daily *prices* from a set of assets
benchmark = (
    expected_returns.returns_from_prices(daily_benchmark)
    .reindex(prices.index)  # benchmark and prices should have matching indices
    .ffill()  # use last available value for benchmark for days where we have prices but no availabe benchmark value
    .values[1:]  # when prices are converted to returns, the first index is lost
)
```

What do you think? Is it a good use case?

**Comment by robertmartin8** [comment]
Hi Felipe,

Great spot! Instead of the annual risk-free rate, we should be using daily. 

So instead of 2%, and assuming 252 trading days in a year, I need it to be `1.02 ^ (1/252) - 1 =  0.000079`. Will fix that now.

Your code makes sense, but at least within the package, I'm going to default to the risk-free rate rather than the market as a benchmark, just because that's what is in the literature.

Thanks for all your help!

**Comment by robertmartin8** [comment]
Fixed by a0c69ef64a5b52894a47d8ff866cec26e5c339eb

**Comment by schneiderfelipe** [comment]
> Hi Felipe,
> 
> Great spot! Instead of the annual risk-free rate, we should be using daily.
> 
> So instead of 2%, and assuming 252 trading days in a year, I need it to be `1.02 ^ (1/252) - 1 = 0.000079`. Will fix that now.
> 
> Your code makes sense, but at least within the package, I'm going to default to the risk-free rate rather than the market as a benchmark, just because that's what is in the literature.
> 
> Thanks for all your help!

Perfect! Thank *you*, this package is fantastic :)
